xdr3/decode.go

```diff
@@ -770,6 +776,37 @@ func (d *Decoder) decodeInterface(v reflect.Value) (int, error) {
 	return d.decode(ve, 0)
 }
 
+func getUnmarshaler(ve reflect.Value) Unmarshaler {
+	// Check the type first to avoid an unnecessary allocation when casting to interface
```

Comment:
Is type switch also allocating? I wonder if can avoid using `reflect` (which usually is slow). See: https://stackoverflow.com/a/28027945/9812359 maybe we can add a small benchmark to check it out. I think, it can simplify the code a bit.

Reply:
From the tests I performed, I don't think it allocates.